### ISSUE TITLE
Tracking: base ledger prototype version 3

### ISSUE THREAD

**Issue by tzemanovic** [anoma/anoma#125]
This is tracking issue for the base ledger prototype version 3 (a follow-up to base ledger prototype version 2 #62).

Goals:

* integrate Ferveo/ABCI++ (blocked on tendermint release)
  * #357 
  * #358 
  * #362 
  * #392 
* IBC integration for VPs/tx [ICS20](https://github.com/informalsystems/ibc-rs/tree/cfbb7663d5bd4767db3646d31f6e7c6258644332/modules/src/application/ics20_fungible_token_transfer)
  * #326 
  * #327 
  * #411 
  * #480 
  * #498 
  * #607 
* PoS 
  * #434
  * #435
  * #436
  * #437
  * #438
  * #439
  * #440
  * #441
  * #442
  * #443
  * #459 
  * #460
  * #462
  * #463
  * #635 
  * #637
  * #638
  * #639
  * #640
* MASP integration - should start with a exploration/design doc
* base ledger
  * #675
  * #674
  * #645
  * storage data schema (described in https://github.com/heliaxdev/anoma-prototype/pull/126)
  * #208
  * #258 
  * #288
  * #4 
  * #298
  * #167 
  * #625 
  * #88 
  * #113 
  * #189 
  * #191 
  * #315
  * #335
  * #634
  * #455 
  * #472
  * #581 
  * #583 
  * #592 
  * #602 
  * #624 
  * #625 
  * #628 
  * #634
  * #641
* WASM
  * #678 
  * #309 
  * #79 
  * #157 and #677 
  * #21 
  * #308
  * #22
  * #564 
* validity predicates
  * #25
  * #193 
  * #495 
  * #636 
* crypto
  * #163 
  * #162 
  * #646 
  * #225 
* storage
  * #631
  * #632, #591
  * #633
  * #582 
  * #7
  * #81 
  * #372 
  * #383 
  * #626 
* gas & fees
  * #72 
  * #293 
* continuous
  * #739
  * #740 
  * updating dependencies
  * updating tooling
  * #481 
* lower priority:
  * #676 
  * #578
  * #208
  * #250 
  * #351 
  * #352 
  * #354 
  * #403 
  * #427 
  * #455 
  * #525

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201757896515792/1201762920590875) by [Unito](https://www.unito.io)

**Comment by tzemanovic** [comment]
I'm updating this issue

**Comment by tzemanovic** [comment]
intent gossiper & matchmaker issues have been separated into #620

**Comment by tzemanovic** [comment]
this workflow is replaced by project board https://github.com/orgs/anoma/projects/1
